src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -526,4 +526,32 @@ private boolean userHasPermission(Course course) {
 
         return ResponseEntity.ok().body(course);
     }
+
+    /**
+     * GET  /courses/:courseId/categories : Returns all categories used in a course
+     *
+     * @param courseId the id of the course to get the categories from
+     * @return the ResponseEntity with status 200 (OK) and the list of categories or with status 404 (Not Found)
+     */
+    @GetMapping(value = "/courses/{courseId}/categories")
+    @PreAuthorize("hasAnyRole('TA', 'INSTRUCTOR', 'ADMIN')")
```

Comment:
Students do not see the categories?

Reply:
students don't need to see the whole list. they see the categories inside the exercise model